contracts/libraries/Safe150Migration.sol

```diff
@@ -0,0 +1,186 @@
+     */
+    function checkGuard() private view {
+        address guard = getGuard();
+
+        if (guard != address(0)) {
+            require(Guard(guard).supportsInterface(type(Guard).interfaceId), "GS300");
+        }
+    }
+
+    /**
+     * @notice Migrate to Safe 1.5.0 Singleton (L1) at `SAFE_150_SINGLETON`
+     * @dev This function should only be called via a delegatecall to perform the upgrade.
+     */
+    function migrateSingleton() public {
+        require(address(this) != MIGRATION_SINGLETON, "Migration should only be called via delegatecall");
+
+        checkGuard();
+
+        singleton = SAFE_150_SINGLETON;
+        emit ChangedMasterCopy(singleton);
+    }
+
+    /**
+     * @notice Migrate and set the fallback handler to Safe 1.5.0 Compatibility Fallback Handler.
+     */
+    function migrateWithFallbackHandler() public {
+        migrateSingleton();
+
+        ISafe(address(this)).setFallbackHandler(SAFE_150_FALLBACK_HANDLER);
+    }
+
+    /**
+     * @notice Migrate and set the guard to the specified address.
+     * @param guard The address of the new guard contract.
+     */
+    function migrateWithSetGuard(address guard) public {
+        require(address(this) != MIGRATION_SINGLETON, "Migration should only be called via delegatecall");
+
+        singleton = SAFE_150_SINGLETON;
+        emit ChangedMasterCopy(singleton);
+
+        ISafe(address(this)).setGuard(guard);
+    }
+
+    /**
+     * @notice Migrate, set the guard to the specified address, and set the fallback handler to Safe 1.5.0 Compatibility Fallback Handler.
+     * @param guard The address of the new guard contract.
+     */
+    function migrateWithSetGuardAndFallbackHandler(address guard) public {
+        migrateWithSetGuard(guard);
+
+        ISafe(address(this)).setFallbackHandler(SAFE_150_FALLBACK_HANDLER);
+    }
+
+    /**
+     * @notice Migrate to Safe 1.5.0 Singleton (L2) at `SAFE_150_SINGLETON_L2`
+     * @dev This function should only be called via a delegatecall to perform the upgrade.
+     */
+    function migrateL2Singleton() public {
+        require(address(this) != MIGRATION_SINGLETON, "Migration should only be called via delegatecall");
+
+        checkGuard();
+
+        singleton = SAFE_150_SINGLETON_L2;
+        emit ChangedMasterCopy(singleton);
+    }
+
+    /**
+     * @notice Migrate to Safe 1.5.0 Singleton (L2) and set the fallback handler to Safe 1.5.0 Compatibility Fallback Handler.
+     */
+    function migrateL2WithFallbackHandler() public {
+        migrateL2Singleton();
+
+        ISafe(address(this)).setFallbackHandler(SAFE_150_FALLBACK_HANDLER);
+    }
+
+    /**
+     * @notice Migrate to Safe 1.5.0 Singleton (L2) and set the guard to the specified address.
+     * @param guard The address of the new guard contract.
+     */
+    function migrateL2WithSetGuard(address guard) public {
+        require(address(this) != MIGRATION_SINGLETON, "Migration should only be called via delegatecall");
```

Comment:
Do I assume correctly that you do not use `migrateL2Singleton()` to avoid the call to `checkGuard();`

Reply:
Yes, that's correct